modules/generative-openai/clients/openai.go

```diff
@@ -71,22 +72,46 @@ func (v *openai) GenerateAllResults(ctx context.Context, textProperties []map[st
 func (v *openai) Generate(ctx context.Context, cfg moduletools.ClassConfig, prompt string) (*ent.GenerateResult, error) {
 	settings := config.NewClassSettings(cfg)
 
-	body, err := json.Marshal(generateInput{
-		Prompt:           prompt,
-		Model:            settings.Model(),
-		MaxTokens:        settings.MaxTokens(),
-		Temperature:      settings.Temperature(),
-		FrequencyPenalty: settings.FrequencyPenalty(),
-		PresencePenalty:  settings.PresencePenalty(),
-		TopP:             settings.TopP(),
-	})
-	if err != nil {
-		return nil, errors.Wrapf(err, "marshal body")
+	var body []byte
+	var err error
+	var oaiUrl string
+
+	if settings.IsLegacy() {
+		oaiUrl, err = url.JoinPath(v.host, "/v1/completions")
```

Comment:
you can use `"path"` package instead of `"url"`, then this part would get simpler:
```
oaiUrl = path.Join(v.host, "/v1/completions")
```